django/template/base.py

```diff
@@ -738,9 +740,21 @@ def resolve(self, context, ignore_failures=False):
             if getattr(func, 'is_safe', False) and isinstance(obj, SafeData):
                 obj = mark_safe(new_obj)
             elif isinstance(obj, EscapeData):
-                obj = mark_for_escaping(new_obj)
+                with warnings.catch_warnings():
+                    # Ignore mark_for_escaping deprecation as this will be
+                    # removed in Django 2.0.
+                    warnings.simplefilter('ignore', category=RemovedInDjango20Warning)
+                    obj = mark_for_escaping(new_obj)
+                    escape_isnt_last_filter = False
             else:
                 obj = new_obj
+        if not escape_isnt_last_filter:
```

Comment:
The tests would have caught that bug. It works because `isinstance(obj, EscapeData)` is checking the value of the previous iteration of the loop, not the current iteration.